OpenRA.Mods.Common/Lint/CheckSequences.cs

```diff
@@ -13,171 +13,136 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
+using OpenRA.Graphics;
 using OpenRA.Mods.Common.Traits.Render;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Lint
 {
-	class CheckSequences : ILintMapPass
+	class CheckSequences : ILintRulesPass
 	{
-		Action<string> emitError;
-
-		List<MiniYamlNode> sequenceDefinitions;
-
-		public void Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Map map)
+		void ILintRulesPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Ruleset rules)
 		{
-			if (map.SequenceDefinitions == null)
-				return;
-
-			this.emitError = emitError;
-
-			sequenceDefinitions = MiniYaml.Load(map, modData.Manifest.Sequences, map.SequenceDefinitions);
+			// Custom maps define rules.Sequences, default mod rules leave it null.
+			if (rules.Sequences == null)
+			{
+				foreach (var kv in modData.DefaultSequences)
+				{
+					Console.WriteLine("Testing default sequences for {0}", kv.Key);
+					Run(emitError, emitWarning, rules, kv.Value);
+				}
+			}
+			else if (!modData.DefaultSequences.Values.Contains(rules.Sequences))
+				Run(emitError, emitWarning, rules, rules.Sequences);
+		}
 
-			var rules = map.Rules;
+		void Run(Action<string> emitError, Action<string> emitWarning, Ruleset rules, SequenceProvider sequences)
+		{
 			var factions = rules.Actors["world"].TraitInfos<FactionInfo>().Select(f => f.InternalName).ToArray();
-			var sequenceProviders = new[] { rules.Sequences };
-
 			foreach (var actorInfo in rules.Actors)
 			{
-				foreach (var renderInfo in actorInfo.Value.TraitInfos<RenderSpritesInfo>())
+				// Actors may have 0 or 1 RenderSprites traits
+				var renderInfo = actorInfo.Value.TraitInfoOrDefault<RenderSpritesInfo>();
+				if (renderInfo == null)
+					continue;
+
+				var images = new HashSet<string>()
 				{
-					foreach (var faction in factions)
-					{
-						foreach (var sequenceProvider in sequenceProviders)
-						{
-							var image = renderInfo.GetImage(actorInfo.Value, sequenceProvider, faction);
-							if (sequenceDefinitions.All(s => s.Key != image.ToLowerInvariant()))
-								emitError("Sprite image {0} from actor {1} using faction {2} has no sequence definition."
```

Comment:
Seems like we are dropping this. When I remove `silo.atreides` from the sequences no error is reported.

Reply:
It turns out that the utility was correct in not reporting this as an error - the FactionImage code had an unexpected fallback to the default image if it referenced an image that wasn't defined.

Removing that workaround now generates the linter error and crashes the game as expected.